File: java-source/base/src/main/java/net/corda/examples/oracle/flow/SignPrime.java
package net.corda.examples.oracle.flow;

import net.corda.core.crypto.TransactionSignature;
import net.corda.core.flows.FlowException;
import net.corda.core.flows.FlowLogic;
import net.corda.core.flows.InitiatingFlow;
import net.corda.core.identity.Party;
import net.corda.core.transactions.FilteredTransaction;

// Simple flow which takes a filtered transaction (exposing only a command containing the nth prime data) and returns
// a digital signature over the transaction Merkle root.
@InitiatingFlow
class SignPrime extends FlowLogic<TransactionSignature> {
    private final Party oracle;
    private final FilteredTransaction ftx;

    public SignPrime(Party oracle, FilteredTransaction ftx) {
        this.oracle = oracle;
        this.ftx = ftx;
    }

    @Override public TransactionSignature call() throws FlowException {
        return sendAndReceive(TransactionSignature.class, oracle, ftx).unwrap(it -> it);
    }
}

File: kotlin-source/base/src/main/kotlin/net/corda/examples/oracle/flow/SignPrime.kt
package net.corda.examples.oracle.flow

import co.paralleluniverse.fibers.Suspendable
import net.corda.core.crypto.TransactionSignature
import net.corda.core.flows.FlowLogic
import net.corda.core.flows.InitiatingFlow
import net.corda.core.identity.Party
import net.corda.core.transactions.FilteredTransaction
import net.corda.core.utilities.unwrap

// Simple flow which takes a filtered transaction (exposing only a command containing the data on
// the nth prime) and returns a digital signature over the transaction's Merkle root.
@InitiatingFlow
class SignPrime(val oracle: Party, val ftx: FilteredTransaction) : FlowLogic<TransactionSignature>() {
    @Suspendable override fun call() = sendAndReceive<TransactionSignature>(oracle, ftx).unwrap { it }
}